### ISSUE TITLE
server.scm test fails

### ISSUE THREAD

**Issue by lantw44** [artyom-poptsov/guile-ssh#10]
It was found on Arch Linux and it seems to be always reproducible:
```
make[3]: Entering directory '/build/guile-ssh/src/guile-ssh-0.11.3/tests'
PASS: log.scm
FAIL: server.scm
PASS: session.scm
PASS: client-server.scm
PASS: popen.scm
PASS: shell.scm
PASS: server-client.scm
PASS: sssh-ssshd.scm
PASS: key.scm
PASS: tunnel.scm
PASS: dist.scm
============================================================================
Testsuite summary for Guile-SSH 0.11.3
============================================================================
# TOTAL: 11
# PASS:  10
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to poptsov.artyom@gmail.com
============================================================================
make[3]: *** [Makefile:653: test-suite.log] Error 1
```

The libssh version used to run the test is 0.8.2. [test-suite.log](https://github.com/artyom-poptsov/guile-ssh/files/2346050/test-suite.log)

**Comment by artyom-poptsov** [comment]
Hello,

could you please send me the following files from `tests` directory after the test fails?
- `server.log`
- `server-libssh.log`
- `server-errors.log`

Thanks,

Artyom

**Comment by lantw44** [comment]
[server.log](https://github.com/artyom-poptsov/guile-ssh/files/2350637/server.log)
[server-libssh.log](https://github.com/artyom-poptsov/guile-ssh/files/2350636/server-libssh.log)
server-errors.log is empty.

**Comment by pushcx** [comment]
I also had this error, running against `libssh 0.8.3-1`.

I narrowed it to `(test-assert-with-log "server-set!, valid values"
` in `tests/server.scm`. Specifically, these lines:

```
                    ,(if (= %libssh-minor-version 7)
                         (list 'hostkey %rsakey %dsakey)
                         '(hostkey "ssh-rsa" "ssh-dss"))
```

My scheme knowledge is very rusty and I don't know this library at all, but the test passed if I deleted these lines or replaced them with `,(list 'hostkey %rsakey %dsakey)`

Perhaps this conditional should check if `libssh-minor-version` is `>=` instead of `=`?
